bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/cocoa/org/eclipse/swt/dnd/DropTarget.java

```diff
@@ -115,30 +115,47 @@ public class DropTarget extends Widget {
 	static final String DEFAULT_DROP_TARGET_EFFECT = "DEFAULT_DROP_TARGET_EFFECT"; //$NON-NLS-1$
 	static final String IS_ACTIVE = "org.eclipse.swt.internal.isActive"; //$NON-NLS-1$
 
+	public void childcalculation(Control c, long cls){
+	if(c instanceof Composite) {
+  	Control[] cal = ((Composite) c).getChildren();
+
+  	for (Control child : cal)
+  	{	if (cal==null)
+  			continue;
+  		else if ((child instanceof Composite)||(child instanceof ToolBar)|| (child instanceof Control)) {
+  					if(child instanceof Label)	{
+  						long imageView = 0;
+  						if ((imageView = OS.objc_msgSend(child.view.id, OS.sel_getImageView)) != 0) {
+  							cls = OS.object_getClass(imageView);
+  							addDragHandlers(cls);  }}
+      			childcalculation(child,cls);
+  		}}}
+	else if (c instanceof Label){
```

Comment:
should this check be inside the `for` loop?

Reply:
updated.